flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1984,6 +1988,17 @@ IntrinsicLibrary::genChar(mlir::Type type,
   return fir::CharBoxValue{cast, len};
 }
 
+// COMMAND_ARGUMENT_COUNT
+fir::ExtendedValue
+IntrinsicLibrary::genArgumentCount(mlir::Type resultType,
+                                   llvm::ArrayRef<fir::ExtendedValue> args) {
+  assert(args.size() == 0);
+  if (!resultType.isa<mlir::IntegerType>())
+    mlir::emitError(loc, "Return type not an INTEGER of default kind");
```

Comment:
The test is not exactly for the reported condition. To get the FIR type for the INTEGER of default kind requires
```c++
auto ty = builder.getIntegerType(builder.getKindMap().
                   getIntegerBitsize(builder.getKindMap().defaultIntegerKind()));
if (resultType != ty) ...
```

Reply:
Thanks. Fixed in last commit.